### PR TITLE
Implement Automatic AzureRM Service Connection

### PR DESCRIPTION
**Pull request by logachev**
## All Submissions:
-------------------------------------
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open PRs for the same update/change?

## What is the current behavior?
-------------------------------------
Issue Number: N/A

## What is the new behavior?
-------------------------------------
Only manual AzureRM service connection is supported.

## Does this introduce a breaking change?
-------------------------------------
- [YES]

It introduces breaking change in the schema to unify manual and automatic service connection resources.

- removed `azure_scope` parameter and computing it on the fly
- moved manual SP credentials to `credentials` block

## Any relevant logs, error output, etc?
-------------------------------------
NA

## Other information
-------------------------------------

### CODE REVIEW

**Comment by nmiodice** [comment]
@logachev thanks for the contribution. Before we start the review process can you fix the build? i believe there is a failing lint check. Running `find . -name "*.go" | xargs gofmt -s -w` should fix:

```
[INFO] Formatting Go Files... If this fails, run 'find . -name "*.go" | xargs gofmt -s -w' to fix
[INFO] Found 1 with formatting issues...
```

File: azuredevops/resource_serviceendpoint_azurerm_test.go [hunk 1]
```diff
@@ -186,27 +246,41 @@ func TestAccAzureDevOpsServiceEndpointAzureRm_CreateAndUpdate(t *testing.T) {
...
 				),
 			}, {
 				Config: testhelper.TestAccServiceEndpointAzureRMResource(projectName, serviceEndpointNameSecond),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckServiceEndpointAzureRMResourceExists(serviceEndpointNameSecond),
 					resource.TestCheckResourceAttrSet(tfSvcEpNode, "project_id"),
-					resource.TestCheckResourceAttrSet(tfSvcEpNode, "azurerm_spn_clientid"),
-					resource.TestCheckResourceAttr(tfSvcEpNode, "azurerm_spn_clientsecret", ""),
 					resource.TestCheckResourceAttrSet(tfSvcEpNode, "azurerm_spn_tenantid"),
-					resource.TestCheckResourceAttrSet(tfSvcEpNode, "azurerm_spn_clientsecret_hash"),
```
Comment (nmiodice) [inline azuredevops/resource_serviceendpoint_azurerm_test.go:206]: Are we making these checks less strict? I may be missing something so apologies if I am, but it looks like we are not testing for evidence of the secret hash here, nor are we asserting that the secret is indeed missing from the state.

Comment (logachev) [reply azuredevops/resource_serviceendpoint_azurerm_test.go:206]: Good catch, thanks! I think it was some leftover..

**Comment by xuzhang3** [comment]
@logachev 
Test with Terraform official ```make``` command failed.
1. error with command ```make lint```, lint check failed
2. error with command ```make test```
```
--- FAIL: TestAzureDevOpsServiceEndpointAzureRM_ExpandFlatten_Roundtrip (0.00s)
panic: Invalid address to set: []string{"creationMode"} [recovered]
        panic: Invalid address to set: []string{"creationMode"}

goroutine 125 [running]:
testing.tRunner.func1.1(0x122fac0, 0xc0002f0fa0)
        D:/tools/Go/src/testing/testing.go:941 +0x3d7
testing.tRunner.func1(0xc0001e2240)
        D:/tools/Go/src/testing/testing.go:944 +0x400
panic(0x122fac0, 0xc0002f0fa0)
        D:/tools/Go/src/runtime/panic.go:967 +0x16b
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*ResourceData).Set(0xc0003d24d0, 0x140bf4e, 0xc, 0x11ce600, 0xc0002f0f70, 0xc00048f940, 0xc0001e4dc0)
        C:/Users/xuzhang3/GolandProjects/terraform-provider-azuredevops/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource_data.go:206 +0x2f5
github.com/microsoft/terraform-provider-azuredevops/azuredevops.flattenServiceEndpointAzureRM(0xc0003d24d0, 0xc000079df0, 0x2039df0)
        C:/Users/xuzhang3/GolandProjects/terraform-provider-azuredevops/azuredevops/resource_serviceendpoint_azurerm.go:116 +0x1af
github.com/microsoft/terraform-provider-azuredevops/azuredevops.TestAzureDevOpsServiceEndpointAzureRM_ExpandFlatten_Roundtrip(0xc0001e2240)
        C:/Users/xuzhang3/GolandProjects/terraform-provider-azuredevops/azuredevops/resource_serviceendpoint_azurerm_test.go:113 +0x122
```

**Comment by logachev** [comment]
@xuzhang3 thanks for pointing it out, fixed.

**Comment by xuzhang3** [comment]
> @xuzhang3 thanks for pointing it out, fixed.

LGTM. Thanks for your contribute

**Pull request merged**